compiler/javac-wrapper/src/org/jetbrains/kotlin/components/JavacBasedJavaResolverCache.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.components
+
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.ConstructorDescriptor
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.descriptors.SimpleFunctionDescriptor
+import org.jetbrains.kotlin.load.java.components.AbstractJavaResolverCache
+import org.jetbrains.kotlin.load.java.structure.JavaClass
+import org.jetbrains.kotlin.load.java.structure.JavaElement
+import org.jetbrains.kotlin.load.java.structure.JavaField
+import org.jetbrains.kotlin.load.java.structure.JavaMethod
+import org.jetbrains.kotlin.resolve.lazy.ResolveSession
+
+class JavacBasedJavaResolverCache(resolveSession: ResolveSession) : AbstractJavaResolverCache(resolveSession) {
+
+    override fun recordMethod(method: JavaMethod, descriptor: SimpleFunctionDescriptor) {}
+
+    override fun recordConstructor(element: JavaElement, descriptor: ConstructorDescriptor) {}
+
+    override fun recordField(field: JavaField, descriptor: PropertyDescriptor) {}
+
+    override fun recordClass(javaClass: JavaClass, descriptor: ClassDescriptor) {}
```

Comment:
I'm not sure these four methods should be empty. This place should be discussed; I'm afraid we can lose some IDE features here. Should be checked attentively who uses the slices `CONSTRUCTOR` / `VARIABLE` / `CLASS` / `FUNCTION`